tlsoverridefirefox/tlsoverridefirefoxsync/firefoxoverridesync.go

```diff
@@ -0,0 +1,132 @@
+package tlsoverridefirefoxsync
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/hlandau/xlog"
+	"github.com/namecoin/ncdns/tlsoverridefirefox"
+	"gopkg.in/hlandau/easyconfig.v1/cflag"
+)
+
+var flagGroup = cflag.NewGroup(nil, "tlsoverridefirefox")
+var syncEnableFlag = cflag.Bool(flagGroup, "sync", false,
+	"Synchronize TLSA records from the Namecoin zone to Firefox's "+
+		"cert_override.txt")
+var firefoxProfileDirFlag = cflag.String(flagGroup, "profiledir", "",
+	"Firefox profile directory")
+
+var log, Log = xlog.New("ncdns.tlsoverridefirefoxsync")
+
+var zoneData string
+var zoneDataReady = false
+var zoneDataMux sync.Mutex
+
+// Note: the reason for the Fatal reaction to errors is that, if we stop
+// syncing the override list, Firefox will continue trusting .bit certs that
+// might be revoked in Namecoin.  Therefore, it is important that, in such a
+// situation, .bit domains must stop resolving until the issue is corrected.
+// Forcing ncdns to exit is the least complex way to achieve this.
+
+func watchZone() {
+	for {
+		// ncdumpzone will load the RPC address/auth from its config
+		// file.
+		cmdDumpZone := exec.Command("ncdumpzone",
+			"--ncdumpzone.format", "firefox-override")
+		zoneOverrides, err := cmdDumpZone.Output()
+		log.Fatale(err, "Couldn't dump zone for Firefox override sync")
```

Comment:
don't do this; refactor ncdumpzone as a library and call it from here and the command line tool

Reply:
Fixed, pending merge of https://github.com/namecoin/ncdns/pull/82 .